Sources/Common/DIGraph.swift

```diff
@@ -22,10 +22,25 @@ public class DIGraph {
      DIGraph.shared.override(mockOffRoadWheels, OffRoadWheels.self)
      ```
      */
-    public func override<Value: Any>(value: Value, forType type: Value.Type) {
+    public func override<T: Any>(value: T, forType type: T.Type) {
         overrides[String(describing: type)] = value
     }
 
+    // Retrieves an overridden instance of a specified type from the `overrides` dictionary.
+    // If an overridden instance exists and can be cast to the specified type, it is returned; otherwise, nil is returned.
+    public func getOverriddenInstance<T: Any>() -> T? {
+        // Get the type name as the key for the dictionary.
+        let typeName = String(describing: T.self)
+
+        // Attempt to get and cast the overridden instance from the dictionary.
+        if let overriddenInstance = overrides[typeName] as? T {
+            return overriddenInstance
+        }
+
+        // If the instance is not found or cannot be cast, return nil.
+        return nil
+    }
+
```

Comment:
I suggest we remove the optional casting code from this function and instead do a force cast, `as!`. 

This function is only called during automated tests and the force cast will only fail if the developer made a mistake setting up the dependency override. If tests crash, it's easy for the developer to catch the override issue and fix it. Otherwise, the override could go unnoticed and cause tests to be much harder to debug.

Reply:
hmm, can do that, even though I would prefer logging an error than throwing it.  
We actually use the `getOverriddenInstance` in the sample app at times to override the logger etc. But open to change.

Reply:
🤔 I think it would be a good idea to add a comment to the `overrides` property saying that it's suggested to not use overrides in the SDK source code but only in tests (and sample apps if there is a need). 

If we as a team agree that overriding a dependency should only be used for tests (aka: non production code in the SDK), then doing things such as force casting in a function like this is OK. 

---

Back to your original comment, I agree that a log would be more useful then simply a crash without a log explaining the reason for the crash. 

I would be OK to log a message and then call `fatalError` if the cast is not successful. As long as we cause a crash, I am OK with the solution.

Reply:
Sure, I'll change it.